### PR TITLE
refactor(frontend): enhance email resolution logic and guard email display in confirmation pages

### PR DESCRIPTION
**Pull request by sebcomeau**
### Description

This pull request enhances how email addresses are resolved and displayed throughout the protected and public application flows. The main improvement is stricter validation: user-provided emails are only used if they are valid and verified; otherwise, the system falls back to the client application's contact email. This change is consistently applied in both backend logic and frontend display, and is thoroughly tested.

The most important changes are:

**Email resolution logic:**

* Updated `resolveProtectedStateEmailValue` and `resolvePublicStateEmailValue` to only return the user's email if it is both valid (per `validator.isEmail`) and `emailVerified` is `true`; otherwise, the client application's contact email is used. If neither is available, `undefined` is returned. (`frontend/app/.server/routes/helpers/protected-application-route-helpers.ts` [[1]](diffhunk://#diff-039c8b364a90138a6c482f90803bfe32f2a2f52954c3e634141fd2cee3f68b9dL753-R756) `frontend/app/.server/routes/helpers/public-application-route-helpers.ts` [[2]](diffhunk://#diff-a4df438b0b3a9519e2893cc5e2341f86b025acb957475ddbe4a5d3aff65c27baL633-R636)

**Frontend integration:**

* Refactored confirmation pages (`intake-adult`, `intake-children`, `intake-family`) to use the new email resolution logic, ensuring only valid, verified emails are displayed. Updated variable names and conditional rendering to match the new logic. (`frontend/app/routes/protected/application/intake-adult/confirmation.tsx` [[1]](diffhunk://#diff-43e61c3d1e5b0474431750b376c4358fed8526461a992b3a603351f071789275L86-R86) [[2]](diffhunk://#diff-43e61c3d1e5b0474431750b376c4358fed8526461a992b3a603351f071789275L279-R281) [[3]](diffhunk://#diff-43e61c3d1e5b0474431750b376c4358fed8526461a992b3a603351f071789275L314); `frontend/app/routes/protected/application/intake-children/confirmation.tsx` [[4]](diffhunk://#diff-f329064a804f4b4c6875303594039f5d4a7e2932b6612369a1d518a142b7b7ddL80-R80) [[5]](diffhunk://#diff-f329064a804f4b4c6875303594039f5d4a7e2932b6612369a1d518a142b7b7ddL305-R307) [[6]](diffhunk://#diff-f329064a804f4b4c6875303594039f5d4a7e2932b6612369a1d518a142b7b7ddL340); `frontend/app/routes/protected/application/intake-family/confirmation.tsx` [[7]](diffhunk://#diff-866cca11a9d6f73099e6da33baa9ac520a69c4f56fc0bdd6100e7b577ed3b20aL90-R90) [[8]](diffhunk://#diff-866cca11a9d6f73099e6da33baa9ac520a69c4f56fc0bdd6100e7b577ed3b20aL321-R323) [[9]](diffhunk://#diff-866cca11a9d6f73099e6da33baa9ac520a69c4f56fc0bdd6100e7b577ed3b20aL356)

**Test coverage:**

* Expanded and clarified tests for both email resolution helpers to cover all edge cases: valid/invalid emails, verified/unverified status, and missing client application data. (`frontend/__tests__/.server/routes/helpers/protected-application-route-helpers.test.ts` [[1]](diffhunk://#diff-91254ea35c82f6634ffdd78d092d06ef9ccf3dca89f64d856ab78aea68316056L315-R318) [[2]](diffhunk://#diff-91254ea35c82f6634ffdd78d092d06ef9ccf3dca89f64d856ab78aea68316056R328-R348) [[3]](diffhunk://#diff-91254ea35c82f6634ffdd78d092d06ef9ccf3dca89f64d856ab78aea68316056R358-R373); `frontend/__tests__/.server/routes/helpers/public-application-route-helpers.test.ts` [[4]](diffhunk://#diff-65ecb6ca384b760532fcc240edf60224cc27f62c1d8118bedad58e3f6a857c4fL281-R284) [[5]](diffhunk://#diff-65ecb6ca384b760532fcc240edf60224cc27f62c1d8118bedad58e3f6a857c4fR294-R314) [[6]](diffhunk://#diff-65ecb6ca384b760532fcc240edf60224cc27f62c1d8118bedad58e3f6a857c4fR324-R339)

**Code maintenance:**

* Updated imports to include the `validator` library where needed. (`frontend/app/.server/routes/helpers/protected-application-route-helpers.ts` [[1]](diffhunk://#diff-039c8b364a90138a6c482f90803bfe32f2a2f52954c3e634141fd2cee3f68b9dR8) `frontend/app/.server/routes/helpers/public-application-route-helpers.ts` [[2]](diffhunk://#diff-a4df438b0b3a9519e2893cc5e2341f86b025acb957475ddbe4a5d3aff65c27baR8)
* Cleaned up unused variables and references to the old email property in confirmation pages. (`frontend/app/routes/protected/application/renewal-adult/confirmation.tsx` [frontend/app/routes/protected/application/renewal-adult/confirmation.tsxL91-R91](diffhunk://#diff-ace407a568f33f7da3269a1d2a34692c75c4183826f93972bbbeae4b570bcc2aL91-R91))

These changes ensure greater data integrity and a better user experience by preventing the display or use of invalid or unverified email addresses.

### CODE REVIEW

File: frontend/app/.server/routes/helpers/protected-application-route-helpers.ts [hunk 1]
```diff
@@ -750,8 +751,9 @@ export async function resolveProtectedStateHomeAddressValue(
  * @param state - The protected application state containing the optional email and client application contact information.
  * @returns The resolved email address, or `undefined` if neither is available.
  */
-export function resolveProtectedStateEmailValue(state: PickDeep<ProtectedApplicationState, 'email' | 'clientApplication.contactInformation.email'>): string | undefined {
-  return state.email ?? state.clientApplication?.contactInformation.email;
+export function resolveProtectedStateEmailValue(state: PickDeep<ProtectedApplicationState, 'email' | 'emailVerified' | 'clientApplication.contactInformation.email'>): string | undefined {
+  const hasValidEmailInState = typeof state.email === 'string' && validator.isEmail(state.email) && state.emailVerified === true;
```
[the remark this answers is not in the document]
Comment (sebcomeau) [reply frontend/app/.server/routes/helpers/protected-application-route-helpers.ts:755]: I will update the documentation in my next PR.

File: frontend/app/.server/routes/helpers/public-application-route-helpers.ts [hunk 2]
```diff
@@ -630,8 +631,9 @@ export async function resolvePublicStateHomeAddressValue(
  * @param state - The public application state containing the optional email and client application contact information.
  * @returns The resolved email address, or `undefined` if neither is available.
  */
-export function resolvePublicStateEmailValue(state: Pick<PublicApplicationState, 'email'> & PickDeep<PublicApplicationState, 'clientApplication.contactInformation.email'>): string | undefined {
-  return state.email ?? state.clientApplication?.contactInformation.email;
+export function resolvePublicStateEmailValue(state: PickDeep<PublicApplicationState, 'email' | 'emailVerified' | 'clientApplication.contactInformation.email'>): string | undefined {
+  const hasValidEmailInState = typeof state.email === 'string' && validator.isEmail(state.email) && state.emailVerified === true;
```
[the remark this answers is not in the document]
Comment (sebcomeau) [reply frontend/app/.server/routes/helpers/public-application-route-helpers.ts:635]: I will update the documentation in my next PR.
